src/forges/gitea.ts

```diff
@@ -0,0 +1,180 @@
+      body: options.description,
+      prerelease: options.prerelease,
+    });
+
+    return { releaseLink: release.data.html_url! };
+  }
+
+  async getGitCredentials(): Promise<{
+    email: string;
+    username: string;
+    password: string;
+  }> {
+    return {
+      email: this.email,
+      username: 'oauth',
+      password: this.accessToken,
+    };
+  }
+
+  async getPullRequest(options: {
+    owner: string;
+    repo: string;
+    sourceBranch: string;
+    targetBranch: string;
+  }): Promise<PullRequest | undefined> {
+    const pullRequest = await this.handleApiErrors(
+      this.api.repos.repoGetPullRequestByBaseHead(
+        options.owner,
+        options.repo,
+        options.targetBranch,
+        options.sourceBranch,
+      ),
+    );
+
+    if (!pullRequest?.data) {
+      return undefined;
+    }
+
+    return {
+      // TODO: check if those fields are always present and we can safely use ! here
+      title: pullRequest.data.title!,
+      author: pullRequest.data.user?.login!,
+      number: pullRequest.data.number!,
+      labels: pullRequest.data.labels?.map((label) => label.name!) ?? [],
+      description: pullRequest.data.body!,
+    };
+  }
+
+  async getPullRequestFromCommit(options: {
+    owner: string;
+    repo: string;
+    commitHash: string;
+  }): Promise<PullRequest | undefined> {
+    const pullRequest = await this.handleApiErrors(
+      this.api.repos.repoGetCommitPullRequest(options.owner, options.repo, options.commitHash),
+    );
+
+    if (!pullRequest?.data) {
+      return undefined;
+    }
+
+    return {
+      // TODO: check if those fields are always present and we can safely use ! here
+      title: pullRequest.data.title!,
+      author: pullRequest.data.user?.login!,
+      number: pullRequest.data.number!,
+      labels: pullRequest.data.labels?.map((label) => label.name!) ?? [],
+      description: pullRequest.data.body!,
+    };
+  }
+
+  async addCommentToPullRequest(options: {
+    owner: string;
+    repo: string;
+    pullRequestNumber: number;
+    comment: string;
+  }): Promise<void> {
+    await this.api.repos.issueCreateComment(options.owner, options.repo, options.pullRequestNumber, {
+      body: options.comment,
+    });
+  }
+
+  getRepoUrl(owner: string, repo: string): string {
+    return `${this.url}/${owner}/${repo}`;
+  }
+
+  getCommitUrl(owner: string, repo: string, commitHash: string): string {
+    return `${this.url}/${owner}/${repo}/src/commit/${commitHash}`;
+  }
+
+  getIssueUrl(owner: string, repo: string, issueNumber: string | number): string {
+    return `${this.url}/${owner}/${repo}/issues/${issueNumber}`;
+  }
+
+  getPullRequestUrl(owner: string, repo: string, pullRequestNumber: string | number): string {
+    return `${this.url}/${owner}/${repo}/pulls/${pullRequestNumber}`;
+  }
+
+  getReleaseUrl(owner: string, repo: string, release: string): string {
+    return `${this.url}/${owner}/${repo}/releases/tags/${release}`;
```

Comment:
@pat-s Are you sure its `tags`: https://codeberg.org/woodpecker-plugins/node-pm/releases/tag/1.3.0

Reply:
I thought this references the API endpoint: https://gitea.com/api/swagger#/repository. There it is `tags`. I am currently about to check whether a tag event works with this change (using a custom image). Could be wrong.

Reply:
OK, my bad, it is not an API path but the final release url. Sorry! In this case it's `tag`. Will build locally and try to narrow down the other issues.

Would really like to get this one finished.

Reply:
FYI: https://github.com/go-gitea/gitea/pull/32079

Reply:
This might explain all the 404's in https://ci.codeberg.org/repos/13734/pipeline/15/3, but the 

```
# Creating release
Error: undefined
```

when creating a tag is still unresolved.

Even though inferring the PR from a commit fails, the changelog is created and updated as desired.

Reply:
@pat-s You did not had releases enabled. Works after enabling it: https://ci.codeberg.org/repos/13734/pipeline/16/3 https://codeberg.org/woodpecker-plugins/test-release-plugin/releases/tag/1.0.0

Reply:
🙈

Reply:
Then we're good to go here actually?

Reply:
I think so